File: src/db_connection.ts
import mysql from "mysql2/promise"; // ← /promise är viktigt!

import dotenv from "dotenv";
dotenv.config();



const pool = mysql.createPool({
    host: "localhost",
    user: "root",
    password: process.env.PASSWORD,
    database: "IoT_sensor",
    waitForConnections: true,
    connectionLimit: 10,
    queueLimit: 0,
});

export default pool;

File: tests/db_connection.py
import os
import pytest
import requests
import json 
import mysql.connector # <-- Importera MySQL-drivern

from dotenv import load_dotenv # <-- Importera load_dotenv
# Läs in .env-filen (gör detta i toppen av filen)
load_dotenv()

db_pool = mysql.connector.pooling.MySQLConnectionPool(
    pool_name="iot_pool",
    pool_size=10, # Motsvarar connectionLimit: 10
    host=os.getenv("DB_HOST", "localhost"),
    user=os.getenv("DB_USER", "root"),
    password=os.getenv("DB_PASSWORD"),
    database=os.getenv("DB_NAME", "IoT_sensor_test")
)

def get_connection():
    """Hämtar en anslutning från poolen"""
    return db_pool.get_connection()